### PR TITLE
Update meta.yaml

### PR DESCRIPTION
**Pull request by FelixMoelder**
If there is no objection, I would like to reduce the restriction for xorg-libx11 as only `xorg-libx11=1.7` has been build for `osx-arm64`. Doing so would allow performing a migration for that architecture. 

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by FelixMoelder** [comment]
@conda-forge/xorg-libxext, this is ready for review

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,12 +36,13 @@ requirements:
   host:
     - pthread-stubs
     - xorg-kbproto
-    - xorg-libx11 1.6.*
+    - xorg-libx11 >=1.6,<2
+    - xorg-libx11
```
Comment (pkgw) [inline recipe/meta.yaml:41]: Maybe I'm missing something, but is it correct to have two different lines listing `xorg-libx11` here?

Comment (mbargull) [reply recipe/meta.yaml:41]: Hi @pkgw! Yes, I've told Felix to include it this way.
If we'd (for whatever reason) include `xorg-libx11` in the global pinnings at some point, then `conda-build` will pick up a version-constraint-less entry to apply the pinning to. For entries that have constraints, (like the `>=1.6` here) `conda-build` will not apply (potential) pinnings too.

TL;DR: `- xorg-libx11 >=1.6,<2` is for us to declare a minimum version we want to use for the build and `- xorg-libx11` is for `conda-build` to apply pinnings to.

**Comment by mbargull** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (pkgw) [reply recipe/meta.yaml:41]: Ah, I wasn't aware of that pattern. The intention is certainly non-obivous — would one of you mind adding a comment to the `meta.yaml` briefly explaining what's going on?

Comment (mbargull) [reply recipe/meta.yaml:41]: Happy to!
I'd rather have more than less of such explanatory comments anyway :).
Let me know if you'd want me to clarify things further!

**Review by pkgw — APPROVED** [review]
Thanks, I think this is good now!
